mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -300,6 +318,10 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.2.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
```

Comment:
I think we can not override param in any ```set***``` function, since ML pipeline API supports other param setting method like:
```
def fit(dataset: Dataset[_], paramMap: ParamMap): M = {
    copy(paramMap).fit(dataset)
  }
```
I think the only way to override param is at the start of ```fit``` function.